example-app/SantokuApp/src/features/demo-push-notification/pages/PushNotificationPage.tsx

```diff
@@ -1,30 +1,81 @@
+import type {FirebaseMessagingTypes} from '@react-native-firebase/messaging';
 import messaging from '@react-native-firebase/messaging';
 import axios, {AxiosError} from 'axios';
-import React, {useCallback, useState} from 'react';
+import {log} from 'bases/logging';
+import {Item, SelectPicker} from 'bases/ui/picker/SelectPicker';
+import {SpecAndSourceCodeLink} from 'features/demo-github-link/components/SpecAndSourceCodeLink';
+import {ErrorResponse} from 'features/sandbox/apis/model';
+import React, {useCallback, useEffect, useState} from 'react';
 import {SafeAreaView, ScrollView, StyleSheet, View} from 'react-native';
 import {Button, Divider, Text} from 'react-native-elements';
 
-import {ErrorResponse} from '../../sandbox/apis/model';
 import {getFcmToken} from '../services/getFcmToken';
 import {notifyMessageToAll as callNotifyMessageToAll} from '../services/notifyMessageToAll';
 import {notifyMessageToMe as callNotifyMessageToMe} from '../services/notifyMessageToMe';
 import {openSettings} from '../services/openSettings';
-import {requestUserPermission} from '../services/requestUserPermission';
-import {AuthStatusType} from '../types/AuthStatusType';
+import {requestUserPermission as requestUserPermissionService} from '../services/requestUserPermission';
+
+type AuthStatusType = 'NOT_DETERMINED' | 'DENIED' | 'AUTHORIZED' | 'PROVISIONAL';
+
+const channels = [
+  {value: undefined, label: 'チャンネル指定なし'},
+  {value: 'emergencyChannel', label: '重要度:緊急チャンネル'},
+  {value: 'highChannel', label: '重要度:高チャンネル'},
+  {value: 'middleChannel', label: '重要度:中チャンネル'},
+  {value: 'lowChannel', label: '重要度:低チャンネル'},
+];
 
 export const PushNotificationPage: React.FC = () => {
   const [authStatus, setAuthStatus] = useState<AuthStatusType>();
-
   const [token, setToken] = useState<string>();
 
+  const [channelKey, setChannelKey] = useState<string>();
+  const onSelectedChannelChange = useCallback((selectedItem?: Item<string | undefined>) => {
+    setChannelKey(selectedItem?.value);
+  }, []);
+
+  const getPermissionStatusForDisplay = useCallback((authStatus: FirebaseMessagingTypes.AuthorizationStatus) => {
+    switch (authStatus) {
+      case messaging.AuthorizationStatus.NOT_DETERMINED:
+        return 'NOT_DETERMINED';
+      case messaging.AuthorizationStatus.DENIED:
+        return 'DENIED';
+      case messaging.AuthorizationStatus.AUTHORIZED:
+        return 'AUTHORIZED';
+      case messaging.AuthorizationStatus.PROVISIONAL:
+        return 'PROVISIONAL';
+    }
+  }, []);
+
+  const getPermissionStatus = useCallback(async () => {
+    const authStatus = await messaging().hasPermission();
+    setAuthStatus(getPermissionStatusForDisplay(authStatus));
```

Comment:
authStatus -> permissionStatus ?

Reply:
I had set it to authStatus because the type is AuthorizationStatus, but I changed it to permissionAuthStatus because I think it is a little confusing.
2d2c1f8923f2af7691af99ebfb996b804760d336